qep_333_pyqgis_screenshots.md

```diff
@@ -0,0 +1,245 @@
+    
+    Screenshots should be stored in the dest_path folder.
+    
+    The method should return the corresponding markdown to include in the class
+    header, containing `.. image:: filename.png` tags as desired.
+    
+    Multiple images may be generated and included in the returned markdown. 
+    """
+```
+
+To assist with screenshot generation, a new ``ScreenshotUtils`` class will be added
+to the repository. This class will contain static methods for helping with screenshots,
+e.g.
+
+``` py
+
+class ScreenshotUtils:
+
+    @staticmethod
+    def capture_widget(widget: QWidget, width=300, height=None) -> QImage:
+        """
+        Captures the specified widget, using consistent margins and appearance.
+        
+        Returns the captured QImage.
+        """
+    
+    @staticmethod
+    def capture_combobox_with_dropdown(combo: QComboBox, width=300) -> QImage:
+        """
+        Captures a QComboBox widget with the combo box popup in an expanded state.
+        
+        The resultant capture will be sufficiently high to include all of the combo box entries.
+        
+        Returns the captured QImage.
+        """
+```
+
+## Examples
+
+### CharacterWidget
+
+To generate a screenshot for the [CharacterWidget](https://qgis.org/pyqgis/master/gui/CharacterWidget.html) class,
+the ``screenshots/gui/characterwidget.py`` file will be:
+
+``` py
+from pathlib import Path
+from qgis.gui import CharacterWidget
+from screenshots.utils import ScreenshotUtils
+
+def generate_screenshot(dest_path: Path):
+    widget = CharacterWidget()
+
+    im = ScreenshotUtils.capture_widget(widget, width=490, height=320)
+    im.save((dest_path / 'characterwidget.png').as_posix())
+
+    return ("\n.. image:: characterwidget.png\n")
+```
+
+The resultant documentation page looks like this:
+
+![character_widget.png](images/qep333/character_widget.png)
+
+
+### QgsColorBox
+
+To generate multiple screenshots for the [QgsColorBox](https://qgis.org/pyqgis/master/gui/QgsColorBox.html) class,
+the ``screenshots/gui/qgscolorbox.py`` file will be:
+
+``` py
+from pathlib import Path
+from qgis.gui import QgsColorBox, QgsColorWidget
+from screenshots.utils import ScreenshotUtils
+
+from qgis.PyQt.QtGui import QColor
+
+def generate_screenshot(dest_path: Path):
+    widget = QgsColorBox()
+    widget.setColor(QColor(100, 150, 200))
+
+    im = ScreenshotUtils.capture_widget(widget, width=320, height=320)
+    im.save((dest_path / 'color_box_value.png').as_posix())
+
+    widget.setComponent(QgsColorWidget.ColorComponent.Hue)
+    widget.setColor(QColor(150, 150, 200))
+    im = ScreenshotUtils.capture_widget(widget, width=320, height=320)
+    im.save((dest_path / 'color_box_hue.png').as_posix())
+
+    return ("\nQgsColorBox using the QgsColorWidget.ColorComponent.Value component:\n\n.. image:: color_box_value.png\n" +
+            "\nQgsColorBox using the QgsColorWidget.ColorComponent.Hue component:\n\n.. image:: color_box_hue.png\n")
```

Comment:
I think the return value should be code agnostic, a dict with file name and label.
This is more futureproof and flexible for other usage.
The sphinx part should be then generated in the sphinx code

Reply:
Done